crates/sui-core/src/authority_active/checkpoint_driver/mod.rs

```diff
@@ -324,9 +324,10 @@ where
         .iter()
         .for_each(|(auth, _proposal, checkpoint)| {
             if let AuthenticatedCheckpoint::Signed(signed) = checkpoint {
-                // We check this signature is higher than the highest known checkpoint.
+                // We are interested in this signed checkpoint only if it is
+                // newer than the highest known cert checkpoint.
                 if let Some(newest_checkpoint) = &highest_certificate_cert {
-                    if newest_checkpoint.summary.sequence_number > signed.summary.sequence_number {
+                    if newest_checkpoint.summary.sequence_number >= signed.summary.sequence_number {
```

Comment:
Good catch -- I make this mistake often enough that I wonder if we should re-define the API to check internally given an amount of stake.